### PR TITLE
Added functions for skvh_copy job

### PR DESCRIPTION
**Pull request by shamis**
Added readGELTKeys/4, readGELTKeys/5 to read keys between two keys with a limit

### CODE REVIEW

File: src/imem_dal_skvh.erl [hunk 1]
```diff
@@ -1055,6 +1072,17 @@ readGELT(_User, Channel, DecodedKey1, DecodedKey2, Limit) ->
         true -> [skvh_rec_to_map(R) || R <- L ]
     end.
 
+readGELTKeys(User, Channel, DecodedKey1, DecodedKey2) ->
+    readGELTKeys(User, Channel, DecodedKey1, DecodedKey2, 0).
```
Comment (c-bik) [inline src/imem_dal_skvh.erl]: Limit is mandatory. `readGELTKeys/4` is not required

Comment (shamis) [inline src/imem_dal_skvh.erl]: @c-bik have made this change

File: src/imem_dal_skvh.erl [hunk 2]
```diff
@@ -376,6 +379,8 @@ create_check_channel(Channel) ->
     create_check_channel(Channel, [audit,history]).
 
 -spec create_check_channel(binary()|atom(), [atom()|{atom(),any()}]) -> ok | no_return().
+create_check_channel(UserId, Channel) when UserId == system ->
+    create_check_channel(Channel);
```
Comment (c-bik) [inline src/imem_dal_skvh.erl]: Rewrite:
```erlang
-spec create_check_skvh(ddEntityId(),binary()|atom()) -> ok | no_return().
create_check_skvh(UserId, Channel) ->
    %% TODO : Possible future validation, Ignored for now
    create_check_channel(Channel, []).
```

File: src/imem_dal_skvh.erl [hunk 3]
```diff
@@ -1055,6 +1068,22 @@ readGELT(_User, Channel, DecodedKey1, DecodedKey2, Limit) ->
...
+    {L,_} = imem_meta:select(TableName, [MatchFunction], Limit),
+    [binterm_to_term_key(R) || R <- L].
+
+readGELTHashes(_User, Channel, DecodedKey1, DecodedKey2, Limit) ->
+    TableName = atom_table_name(Channel),
+    Key1 = term_key_to_binterm(DecodedKey1),
+    Key2 = term_key_to_binterm(DecodedKey2),
+    MatchFunction = {?MATCHHEAD, [{'>=', '$1', match_val(Key1)}, {'<', '$1', match_val(Key2)}], ['$_']},
+    {L,_} = imem_meta:select(TableName, [MatchFunction], Limit),
+    [{binterm_to_term_key(Key), Hash} || #skvhTable{ckey=Key, chash=Hash} <- L].
```
Comment (c-bik) [inline src/imem_dal_skvh.erl]: Rewrite:
```erlang
MatchFunction = {?MATCHHEAD, [{'>=', '$1', match_val(Key1)}, {'<', '$1', match_val(Key2)}], [{{'$1','$3'}}]},
 +    {L,_} = imem_meta:select(TableName, [MatchFunction], Limit),
 +    [{binterm_to_term_key(Key), Hash} || {Key, Hash} <- L].
```

File: src/imem_dal_skvh.erl [hunk 4]
```diff
@@ -1044,6 +1049,14 @@ readGE(_User, Channel, DecodedKey, Limit) ->
 	{L,_} = imem_meta:select(TableName, [MatchFunction], Limit),
     [skvh_rec_to_map(R) || R <- L ].
 
+readGELTNoError(_User, Channel, DecodedKey1, DecodedKey2, Limit) ->
+    TableName = atom_table_name(Channel),
+    Key1 = term_key_to_binterm(DecodedKey1),
+    Key2 = term_key_to_binterm(DecodedKey2),
+    MatchFunction = {?MATCHHEAD, [{'>=', '$1', match_val(Key1)}, {'<', '$1', match_val(Key2)}], ['$_']},
+    {L,_} = imem_meta:select(TableName, [MatchFunction], Limit),
+    [skvh_rec_to_map(R) || R <- L ].
```
Comment (c-bik) [inline src/imem_dal_skvh.erl]: Rename `readGELTNoError`-> `readGELTMap`

**Comment by c-bik** [comment]
Add Eunit tests for
`readGELTKeys/5`, `readGELTHashes/5` and `readGELTMap/5`

**Pull request merged**